src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -44,20 +44,31 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             }
             error = null;
 #if CORECLR
-            object obj = JsonConvert.DeserializeObject(input, new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.None, MaxDepth = 1024 });
-
-            // JObject is a IDictionary
-            if (obj is JObject)
+            object obj = null;
+            try
             {
+                obj = JsonConvert.DeserializeObject(input, new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.None, MaxDepth = 1024 });
+
+                // JObject is a IDictionary
                 var dictionary = obj as JObject;
-                obj = PopulateFromJDictionary(dictionary, out error);
+                if (dictionary != null)
```

Comment:
I hope [Declaration Expressions](http://stackoverflow.com/a/33299637) can be brought back in C# 7. It will make code like this much more concise.